### PR TITLE
Added ability to request control queue given a call ID

### PR DESCRIPTION
**Pull request by danielfinke**

### CODE REVIEW

File: applications/ecallmgr/src/ecallmgr_call_control.erl [hunk 1]
```diff
@@ -225,6 +229,12 @@ handle_call_events(JObj, Props) ->
         _Else -> 'ok'
     end.
 
+-spec handle_control_queue_req(wh_json:object(), wh_proplist()) -> 'ok'.
+handle_control_queue_req(JObj, Props) ->
+    Srv = props:get_value('server', Props),
+    put('callid', wh_json:get_value(<<"Call-ID">>, JObj)),
```
Comment (fenollp) [inline applications/ecallmgr/src/ecallmgr_call_control.erl]: Use `wh_util:put_callid/1` (without the wh_json call, even)

**Comment by lazedo** [comment]
@danielfinke what's the use for this ? don't we have it already in whapps_call record ?

**Comment by danielfinke** [comment]
@lazedo The whapps_call has the control_queue if your application caught the route_win for the call. In any other circumstance, you have no idea what the control queue is. For example, we've been working on a presence dashboard that binds to call events and supports firing things such as hangups or eavesdrop by a button click. The whapps_call_command functions for these actions require the control queue of the call, so we fetch it using this PR's request/response

**Comment by lazedo** [comment]
@danielfinke PR to master, reference this one so we can continue discussion.

**Comment by fenollp** [comment]
@lazedo Why, there's no need to re-create a PR if the only thing to do is rebase against master.
Why do you keep closing these?

**Comment by lazedo** [comment]
its not a rebase that's needed, the pr was to merge into 3.21

**Comment by fenollp** [comment]
But maybe Daniel can edit that in the current PR?

**Comment by danielfinke** [comment]
@fenollp it doesn't look like I have that ability

**Pull request closed without merging**